Makefile

```diff
@@ -28,163 +27,157 @@ export IRONIC_INSPECTOR_ENDPOINT=http://localhost:5050/v1/
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
 .PHONY: help
 help:  ## Display this help
 	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 	@echo
 	@echo "Variables:"
-	@echo "  TEST_NAMESPACE   -- project name to use ($(TEST_NAMESPACE))"
-	@echo "  SETUP            -- controls the --no-setup flag ($(SETUP))"
+	@echo "  RUN_NAMESPACE    -- project name to use for run target ($(RUN_NAMESPACE))"
 	@echo "  GO_TEST_FLAGS    -- flags to pass to --go-test-flags ($(GO_TEST_FLAGS))"
 	@echo "  DEBUG            -- debug flag, if any ($(DEBUG))"
 
-.PHONY: $(KUSTOMIZE)
-$(KUSTOMIZE):
-	cd $(TOOLS_DIR); ./install_kustomize.sh
+# Run tests
+test: generate fmt vet manifests unit ## Run common developer tests
 
-.PHONY: test
-test: fmt generate lint vet unit ## Run common developer tests
+.PHONY: unit
+unit: ## Run the unit tests
+	go test $(GO_TEST_FLAGS) ./... -coverprofile cover.out
 
-.PHONY: show_packages show_dirs
-show_packages:
-	@echo $(PACKAGES)
-show_dirs:
-	@echo $(CODE_DIRS)
+# Compatibility alias from older version of this file
+unit-cover: unit
 
-.PHONY: generate
-generate: bin/operator-sdk ## Run the operator-sdk code generator
-	./bin/operator-sdk generate $(VERBOSE) k8s
-	./bin/operator-sdk generate $(VERBOSE) crds --crd-version=v1
-	openapi-gen \
-		--input-dirs ./pkg/apis/metal3/v1alpha1 \
-		--output-package ./pkg/apis/metal3/v1alpha1 \
-		--output-base "" \
-		--output-file-base zz_generated.openapi \
-		--report-filename "-" \
-		--go-header-file /dev/null
+.PHONY: unit-verbose
+unit-verbose: ## Run unit tests with verbose output
+	VERBOSE=-v make unit
 
-bin/operator-sdk: bin
-	make -C tools/operator-sdk install
+# Build manager binary
+manager: generate fmt vet ## Build the primary controller binary
+	go build -ldflags $(LDFLAGS) -o bin/manager main.go
 
-bin:
-	mkdir -p bin
+# Compatibility alias from older version of this file
+build: manager
 
-.PHONY: travis
-travis: unit-verbose lint
+# Run against the configured Kubernetes cluster in ~/.kube/config
+run: generate fmt vet manifests ## Run the controller locally
+	go run -ldflags $(LDFLAGS) ./main.go -dev -namespace $(RUN_NAMESPACE)
 
-.PHONY: unit
-unit: ## Run unit tests
-	go test $(GO_TEST_FLAGS) $(PACKAGES)
+# Run against the configured Kubernetes cluster in ~/.kube/config
+demo: generate fmt vet manifests ## Run the controller in demo mode
+	go run -ldflags $(LDFLAGS) ./main.go -dev -demo-mode -namespace $(RUN_NAMESPACE)
 
-.PHONY: unit-cover
-unit-cover: ## Run unit tests with code coverage
-	go test -coverprofile=$(COVER_PROFILE) $(GO_TEST_FLAGS) $(PACKAGES)
-	go tool cover -func=$(COVER_PROFILE)
+# Install CRDs into a cluster
+install: manifests
+	kustomize build config/crd | kubectl apply -f -
```

Comment:
is kustomize assumed to be installed?

Reply:
We used to have a rule to install it. I've restored that in an update to this PR.